packages/babel-preset-react/src/index.js

```diff
@@ -5,15 +5,25 @@ import transformReactJSXSource from "@babel/plugin-transform-react-jsx-source";
 import transformReactJSXSelf from "@babel/plugin-transform-react-jsx-self";
 
 export default function(context, opts = {}) {
-  const development = opts.development || false;
+  const pragma = opts.pragma || "React.createElement";
+  const pragmaFrag = opts.pragmaFrag || "React.Fragment";
+  const throwIfNamespace =
+    opts.throwIfNamespace === undefined ? true : !!opts.throwIfNamespace;
+  const development = !!opts.development;
+  const useBuiltIns = !!opts.useBuiltIns;
 
   if (typeof development !== "boolean") {
-    throw new Error("Preset react 'development' option must be a boolean.");
+    throw new Error(
+      "@babel/preset-react 'development' option must be a boolean.",
+    );
   }
 
   return {
     plugins: [
-      transformReactJSX,
+      [
+        transformReactJSX,
+        { pragma, pragmaFrag, throwIfNamespace, useBuiltIns },
```

Comment:
Do we need to expose the ones other than `useBuiltins`? Since they are explicitly for things other than React, not sure if we'd expect anyone to use them, and they obfuscate `useBuiltins` a bit.

Reply:
suggested by @Kovensky https://github.com/babel/babel/pull/6733#discussion_r148711145

depends on whether we think people consuming other projects want to use the preset or the plugin, ya dono